staging/kos/pkg/controllers/ipam/ipam.go

```diff
@@ -109,24 +109,22 @@ type IPAMController struct {
 	statusUsedHistogram prometheus.Histogram
 }
 
-// NetworkAttachmentData holds the local state for a
-// NetworkAttachment.  The fields can only be accessed by a worker
-// thread working on the NetworkAttachment.  The data for a given
-// attachment is used to remember a status update while it is in
-// flight. When the attachment's ResourceVersion is either
-// anticipatingResourceVersion or anticiaptedResourceVersion,
-// anticipationSubnetRV is the ResourceVersion of the attachment's
-// subnet, and anticipatedIPv4 != nil then that address has been
-// chosen based on that subnet revision and written into the
-// attachment's status and there exists an IPLock that supports this,
-// even if this controller has not yet been notified about that lock;
-// when any other ResourceVersion is seen these three fields get set
-// to their zero value.
+// NetworkAttachmentData holds the local state for a NetworkAttachment.
+// The fields can only be accessed by a worker thread working on the
+// NetworkAttachment.  The data for a given attachment is used to remember a
+// status update while it is in flight. When the attachment's ResourceVersion
+// is either anticipatingResourceVersion or anticipatedResourceVersion,
+// anticipationSubnetUID is the UID of the attachment's subnet, and
+// anticipatedIPv4 != nil then that address has been chosen based on that
+// subnet and written into the attachment's status and there exists an IPLock
+// that supports this, even if this controller has not yet been notified about
+// that lock; when any other ResourceVersion or UID is seen these three fields
+// get set to their zero value.
 type NetworkAttachmentData struct {
 	anticipatedIPv4             gonet.IP
 	anticipatingResourceVersion string
 	anticipatedResourceVersion  string
-	anticipationSubnetRV        string
+	anticipationSubnetUID       k8stypes.UID
```

Comment:
anticipation data to remember in-flight status updates was reset if the recorded subnet RV did not match the last seen subnet RV. Such data should be reset if a subnet VNI or CIDR change. After immutability such change can happen only if the subnet UID changes => replace subnet RV with subnet UID in anticipation data: there are less useless resets of the anticipation data.